### DISCUSSION TITLE
How to retain Session data in FastAPI the way we retain in Streamlit framework

### DISCUSSION THREAD

**Discussion by revanks** [fastapi/fastapi#8472] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#below is the Streamlit app code

def app():

    st.title("HOME")
    st.markdown("## Data Upload")

    # Upload the dataset and save as csv
    st.markdown("### Upload a csv file.") 

    # Code to read a single file 
    uploaded_file = st.file_uploader("Choose a file", type = ['csv'])
    if uploaded_file.name.endswith(".csv"):
        data = pd.read_csv(uploaded_file)

    st.session_state.data = data 
  """#  Here we are retaining data with session_state for next operation, how can we perform same operation FastAPI..?    """
```

### Description

In the above Streamlit app code  we are retaining data with **session_state**  for next operation, How can we perform same operation in FastAPI..?  

### Operating System

Windows

### Operating System Details

10

### FastAPI Version

0.75.2

### Python Version

Python 3.9.12

**Comment by yinziyan1206** [comment]
Global: just save it as a global object or in your app instance.
Local: use contextvar, or save it in request
